Automattic/wp-calypso
# Update domain only flow: `site-or-domain` step

## Changes proposed in this Pull Request

This PR enables the reskinned design on domain only flow and updates the second step (`site-or-domain`).

It's part of the new domain-only flow (pcYYhz-ye-p2)

![before-logged-in](https://user-images.githubusercontent.com/2797601/150507329-d8a0edba-6634-45b0-90a0-c2eac02d3707.png)

![before-logged-out](https://user-images.githubusercontent.com/2797601/150507344-75a46c1b-9f54-49b8-b515-ed2139c4d41f.png)

![after-logged-in](https://user-images.githubusercontent.com/2797601/151006072-8ee66f49-fa77-4a3a-a021-4d0be426e09e.png)

![after-logged-out](https://user-images.githubusercontent.com/2797601/151006092-826be8dc-edd1-4a4b-b26d-fbb041e410fc.png)

## Testing instructions

- Build this branch locally or open the live Calypso link
- Navigate to to `/start/domain/domain-only`
- Enter a domain name and select it, in order to move to the next step
- Verify that the layout matches the screenshots above (if you are not logged in the option "Existing site" should be hide)
